onnxruntime/core/optimizer/matmul_scale_fusion.cc

```diff
@@ -131,6 +131,11 @@ std::vector<ScaleMergeInfo> GetInputNodeMerges(
     ORT_ENFORCE(input_node.InputDefs().size() == 2 && scale_and_index->second < 2);
     const int to_scale_index = 1 - scale_and_index->second;
 
+    // check if the non-scale input is graph input and scalar 
+    const auto non_scale_node_arg = input_node.InputDefs()[to_scale_index];
+    const auto* shape = non_scale_node_arg->Shape();
+    if (graph_utils::IsGraphInput(graph, non_scale_node_arg) && shape->dim_size() == 0) continue;
```

Comment:
>IsGraphInput [](http://example.com/codeflow?start=21&length=12)

I think we don't need check it is a graph input, any reason for doing this?

Reply:
right, we don't need it